### PR TITLE
Backport hashdeep package from Ubuntu 17.10

### PR DESCRIPTION
**Pull request by scollazo**
This fixes archivematica#873

### CODE REVIEW

File: debs/xenial/hashdeep/Makefile [hunk 1]
```diff
@@ -0,0 +1,37 @@
+NAME          = hashdeep
+VERSION       ?= 4.4
+RELEASE	      ?= 1~16.04
+GIT_URL	      = https://github.com/LibraryOfCongress/bagit-java/
```
Comment (sevein) [inline debs/xenial/hashdeep/Makefile]: Mixed spaces and tabs.

**Comment by sevein** [comment]
@scollazo is this change still relevant? I've seen than https://github.com/artefactual/archivematica/issues/873 was closed.

**Comment by scollazo** [comment]
Yes, a workaround was applied to augustus, but needs to be extended to new installs

**Comment by sevein** [comment]
@scollazo once this is merged how do we guarantee that the package is installed? I see in osdeps that we're only trying to install md5deep.

File: debs/xenial/hashdeep/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
+FROM ubuntu:xenial
+
+ARG GIT_URL
+ARG GIT_BRANCH
+ARG PACKAGE
```
Comment (sevein) [inline debs/xenial/hashdeep/Dockerfile]: Unused?

File: debs/xenial/hashdeep/Makefile [hunk 3]
```diff
@@ -0,0 +1,37 @@
+NAME          = hashdeep
+VERSION       ?= 4.4
+RELEASE       ?= 1~16.04
+GIT_URL       = https://github.com/LibraryOfCongress/bagit-java/
```
Comment (sevein) [inline debs/xenial/hashdeep/Makefile]: Unused?

File: debs/xenial/hashdeep/Makefile [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+DOCKER_VOLUME = "/src"
+DOCKER_IMAGE  = "debbuild-$(NAME)-$(VERSION)"
+
+.PHONY: build-docker-image build deb-build deb-clean deb-test update-changelog
+
+all: build-docker-image build
+
+build-docker-image:
+	@echo "==> Building Docker image with build environment."
+	@docker build --tag "$(DOCKER_IMAGE)" --build-arg GIT_URL="$(GIT_URL)" --build-arg GIT_BRANCH=v"$(VERSION)" --build-arg PACKAGE="$(NAME)" .
```
Comment (sevein) [inline debs/xenial/hashdeep/Makefile]: I've just realized that these build arguments that we're passing are also not being used in the Dockerfile.

File: debs/xenial/hashdeep/Makefile [hunk 5]
```diff
@@ -0,0 +1,36 @@
+NAME          = hashdeep
+VERSION       ?= 4.4
+RELEASE       ?= 1~16.04
+DEB_TOPDIR    = "/debbuild"
+DOCKER_VOLUME = "/src"
+DOCKER_IMAGE  = "debbuild-$(NAME)-$(VERSION)"
+
+.PHONY: build-docker-image build deb-build deb-clean deb-test update-changelog
```
Comment (sevein) [inline debs/xenial/hashdeep/Makefile]: `deb-test` and `update-changelog` aren't defined targets in this document.

File: debs/xenial/hashdeep/Makefile [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+	@echo "==> Build package."
+	cd /debbuild/$(NAME)-$(VERSION) && dpkg-buildpackage 
+	@echo "==> Copying built  files."
+	mkdir -p $(DOCKER_VOLUME)/build/
+	cp  $(DEB_TOPDIR)/$(NAME)_*.* $(DOCKER_VOLUME)/build/ 
+
+deb-clean:
+	@echo "==> Cleaning up previous RPMs builds."
+	@rm -rf $(DOCKER_VOLUME)/build/
+
```
Comment (sevein) [inline debs/xenial/hashdeep/Makefile]: Extra blank link here.

File: debs/xenial/hashdeep/Makefile [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+	@echo "==> Rebuild package."
+	cd /debbuild/$(NAME)-$(VERSION) && sed -i 's/>=10/>=9/g' debian/control
+	@echo "==> Build package."
+	cd /debbuild/$(NAME)-$(VERSION) && dpkg-buildpackage 
+	@echo "==> Copying built  files."
+	mkdir -p $(DOCKER_VOLUME)/build/
+	cp  $(DEB_TOPDIR)/$(NAME)_*.* $(DOCKER_VOLUME)/build/ 
+
+deb-clean:
+	@echo "==> Cleaning up previous RPMs builds."
```
Comment (sevein) [inline debs/xenial/hashdeep/Makefile]: Nitpick: it reads `RPMs`.

Comment (scollazo) [reply debs/xenial/hashdeep/Makefile]: Change added, and pr rebased & merged.

Thanks for your continuous feedback!

**Pull request merged**